### PR TITLE
No new privs

### PR DESCRIPTION
**Pull request by matajoh**
PR adding support for NoNewPrivileges in various policy objects.

### CODE REVIEW

File: pkg/securitypolicy/regopolicy_test.go [hunk 1]
```diff
@@ -93,6 +93,9 @@ func Test_MarshalRego_Policy(t *testing.T) {
 		containers := make([]*Container, len(p.containers))
 		for i, container := range p.containers {
 			containers[i] = container.toContainer()
+			if containers[i].NoNewPrivileges != container.NoNewPrivileges {
+				panic("weird")
```
Comment (SeanTAllen) [inline pkg/securitypolicy/regopolicy_test.go:97]: panic?

File: pkg/securitypolicy/regopolicy_test.go [hunk 2]
```diff
@@ -3766,6 +3752,81 @@ func Test_Fragment_FrameworkSVN_In_Future(t *testing.T) {
 	}
 }
 
+func Test_Rego_EnforceContainerNoNewPrivilegesPolicy_NoMatches(t *testing.T) {
```
Comment (SeanTAllen) [inline pkg/securitypolicy/regopolicy_test.go:3755]: i see these "no matches" tests, where are the matches tests being done?

**Comment by SeanTAllen** [comment]
looks good to me

File: pkg/securitypolicy/regopolicy_test.go (hunk 1 above)
Comment (matajoh) [reply pkg/securitypolicy/regopolicy_test.go:97]: left over from when I was doing some debugging, whoops

File: internal/oci/annotations.go [hunk 3]
```diff
@@ -48,11 +48,23 @@ func ProcessAnnotations(ctx context.Context, s *specs.Spec) (err error) {
 		}
 	}
 
+	// process security context annotations
+	if s.Process != nil && s.Process.NoNewPrivileges {
+		s.Process.NoNewPrivileges = !ParseAllowPrivilegeEscalation(ctx, s)
```
Comment (anmaxvl) [inline internal/oci/annotations.go:53]: you can just directly call ParseAnnotationsBool I think?

File: pkg/securitypolicy/framework.rego [hunk 4]
```diff
@@ -1046,4 +1053,34 @@ errors[fragment_framework_svn_error] {
...
+noNewPrivileges_matches {
+    input.rule == "exec_external"
+    some process in data.external_processes
+    command_ok(process.command)
+    workingDirectory_ok(process.working_dir)
+    noNewPrivileges_ok(process.no_new_privileges)
+}
+
+errors["invalid noNewPrivileges"] {
+    input.rule in ["create_container", "exec_in_container", "exec_external"]
```
Comment (anmaxvl) [inline pkg/securitypolicy/framework.rego:1084]: I don't think exec_external should be here. It gets a shell into UVM directly and isn't tied to a particular container spec

File: internal/oci/annotations.go (hunk 3 above)
Comment (matajoh) [reply internal/oci/annotations.go:53]: The pattern seems to be to wrap it as seen here, I think it is best to keep it this way to adhere to how other annotations are being parsed.

File: pkg/securitypolicy/framework.rego (hunk 4 above)
Comment (matajoh) [reply pkg/securitypolicy/framework.rego:1086]: So I've added enforcement for exec_external and exec_internal in policy, so they need error management as well.

**Pull request closed without merging**